src/components/CCData.js

```diff
@@ -142,57 +166,61 @@ class CCData extends Component {
                             <Form onSubmit={this.handleSubmit}>
                                 <div className="ccInfoTitles">{i18n.t('cardInfo')}</div>
                                 <Form.Control type="text" name="name" placeholder={i18n.t('name')} required className='ccInfoFormPlaceHolder'
-                                    pattern="([\S]+([\s]+[\S]+)+)" title={i18n.t('firstNlast')} onChange={this.handleInputChange}  
+                                    pattern="([\S]+([\s]+[\S]+)+)" title={i18n.t('firstNlast')} onChange={this.handleInputChange} value={this.state.ccinfo.name}
+                                    ref={(input) => this.nameInput = input} 
                                 />
                                 <br/>
                                 <Form.Control type="text" name="number" placeholder={i18n.t('cardNumber')} required className='ccInfoFormPlaceHolder'
                                     pattern="([0-9 ]{19})|([0-9]{16})" title={i18n.t('cardNumber16digit')} value={this.state.ccinfo.number}
-                                    onChange={this.handleInputChange}  
+                                    onChange={this.handleInputChange} ref={(input) => this.numberInput = input} 
```

Comment:
what does "ref=...." do here?

Reply:
Creates a Dom reference so can make the form focus on that specific input. Used when the card number is invalid aor anything like that